### DISCUSSION TITLE
How to add instance and database to edgedb docker

### DISCUSSION THREAD

**Discussion by aa-deet-eeya** [geldata/gel#2867] [Q&A] [answered]
So I want to run edgedb inside a docker container using docker-compose, I followed the example here in https://github.com/edgedb/edgedb-docker/tree/master/tests/compose

Now the question is how do I create additional instances and databases inside the instance? 

I looked through the docs and docker hub, but couldn't find anything. 

Note: Also the commands in the [docker hub](https://hub.docker.com/r/edgedb/edgedb) is wrong, there's no `edgedb authenticate` command mentioned under "How to use this image" section

**Comment by elprans** [comment, marked as answer]
> Now the question is how do I create additional instances 

An EdgeDB docker container _is_ an instance.  If you need more instances, you can spawn more containers.  However, rather than using `docker compose`, we recommend using the [`edgedb project`](https://www.edgedb.com/docs/cli/edgedb_project/index) and [`edgedb instance`](https://www.edgedb.com/docs/cli/edgedb_instance/index) commands, which automate instance setup and authentication.  Here is a good introductory [blog post](https://www.edgedb.com/blog/introducing-edgedb-projects) about `edgedb project`.

> and databases inside the instance?

Connect to the instance with `edgedb` CLI and run `CREATE DATABASE`.

> Note: Also the commands in the [docker hub](https://hub.docker.com/r/edgedb/edgedb) is wrong, there's no `edgedb authenticate` command mentioned under "How to use this image" section

Hm, it seems the `README` isn't syncing correctly.  Thanks for noticing!

**Reply by 1st1** [reply]
>  there's no edgedb authenticate command mentioned under "How to use this image" section

`edgedb authenticate` was renamed to `edgedb instance link`

**Reply by fantix** [reply]
> Connect to the instance with `edgedb` CLI

To use the CLI, one way to leverage docker-compose is to add `edgedb` CLI to one of the client services:

```patch
--- a/tests/compose/dockerfiles/Dockerfile.py
+++ b/tests/compose/dockerfiles/Dockerfile.py
@@ -1,5 +1,7 @@
+FROM edgedb/edgedb-cli:linux-x86_64-nightly AS cli
 FROM python:3.9

+COPY --from=cli /usr/bin/edgedb /usr/bin/edgedb
 COPY /serve_py /app/serve_py
 COPY /certs/ca.pem /app/ca.pem
 RUN pip install --no-cache-dir -r /app/serve_py/requirements.txt
```

And then you could run the following command to use the CLI while your compose services are running:

```
$ docker-compose run --rm python edgedb --host edgedb --tls-ca-file=/app/ca.pem
```
